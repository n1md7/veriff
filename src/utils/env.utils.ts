import dotenv from 'dotenv';
dotenv.config();

export class EnvUtils {
  static getServerPort() {
    return Number(process.env.PORT) || 3000;
  }

  static getVeriffBaseUrl() {
    const url = process.env.VERIFF_API_BASE_URL;
    if (!url) throw new Error('"VERIFF_API_BASE_URL" is missing');

    return url;
  }

  static getVeriffSessionUUID() {
    const uuid = process.env.VERIFF_SESSION_UUID;
    if (!uuid) throw new Error('"VERIFF_SESSION_UUID" is missing');

    return uuid;
  }
}
