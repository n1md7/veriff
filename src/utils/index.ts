export * from './string.utils';
export * from './response.utils';
export * from './env.utils';
