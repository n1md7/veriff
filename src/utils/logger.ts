export class Logger {
  private static message(msg: unknown, ctx: string, type: string) {
    console.log(`${type} - ${process.pid} - [${Date.now()}] [${ctx}]: ${JSON.stringify(msg)}`);
  }

  static log(msg: unknown, ctx: string) {
    Logger.message(msg, ctx, 'LOG');
  }

  static error(msg: unknown, ctx: string) {
    Logger.message(msg, ctx, 'ERROR');
  }

  static warn(msg: unknown, ctx: string) {
    Logger.message(msg, ctx, 'WARN');
  }
}
