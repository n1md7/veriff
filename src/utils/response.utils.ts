export class ResponseUtils {
  static getResponse(code, data) {
    const random = Math.floor(Math.random() * 10) + 1;
    if (random > 8) {
      return [500, 'There is something wrong with the service'];
    }

    return [code, data];
  }
}
