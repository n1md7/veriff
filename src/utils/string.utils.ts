import { EnvUtils } from './env.utils';

export const findMatch = (haystack: string, needle = EnvUtils.getVeriffSessionUUID()) => {
  return haystack.includes(needle);
};
