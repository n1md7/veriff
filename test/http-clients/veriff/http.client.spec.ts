import { httpClient, VeriffAPI } from '../../../src/http-clients/veriff/http.client';
import { Logger } from '../../../src/utils/logger';

describe('VeriffAPI', () => {
  afterAll(() => jest.clearAllMocks());
  it('should have methods defined', () => {
    const api = new VeriffAPI('');
    expect(api.fetchMediaContext).toBeDefined();
    expect(api.fetchSessionMedia).toBeDefined();
    expect(api.fetchSessionDetails).toBeDefined();
  });

  it('should send api request', async () => {
    const apiCall = jest.spyOn(httpClient, 'get').mockResolvedValue('data');
    const logCall = jest.spyOn(Logger, 'log').mockReturnValue(void 0);

    const api = new VeriffAPI('session-id');
    const reqId = 'req-id';

    await api.fetchSessionMedia(reqId);
    await api.fetchMediaContext(reqId);
    await api.fetchSessionDetails(reqId);

    expect(apiCall).toMatchSnapshot();
    expect(logCall).toMatchSnapshot();
  });
});
