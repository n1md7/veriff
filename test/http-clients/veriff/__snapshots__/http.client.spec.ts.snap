// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VeriffAPI should send api request 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "/sessions/session-id/media",
      Object {
        "headers": Object {
          "x-request-id": "req-id",
        },
      },
    ],
    Array [
      "/media-context/session-id",
      Object {
        "headers": Object {
          "x-request-id": "req-id",
        },
      },
    ],
    Array [
      "/sessions/session-id",
      Object {
        "headers": Object {
          "x-request-id": "req-id",
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Promise {},
    },
    Object {
      "type": "return",
      "value": Promise {},
    },
    Object {
      "type": "return",
      "value": Promise {},
    },
  ],
}
`;

exports[`VeriffAPI should send api request 2`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "Fetching session media",
      "req-id",
    ],
    Array [
      "Fetching media context",
      "req-id",
    ],
    Array [
      "Fetching session details",
      "req-id",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
